UI Pages and themes

ange-02 opened the pull request:
Contains initial UI design for main pantry screen, user settings, pantry users dialog and add ingredient dialog. Also, colors and themes.

farazovo committed:
pages + gradel upgrade

added a outline file called back to add outlines to text boxs
put the code the fragment_recipe menu for menu
put code for join/create panty in dialog create pantry
(unsure if these locations are right)
(if there are any more related fragments that i need to do lmk should be fast)

farazovo committed:
minor fix to a line

ange-02 committed:
UI Pages and themes

Contains initial UI design for main pantry screen, user settings, pantry users dialog and add ingredient dialog.
Also, colors and themes.

ange-02 committed:
Fixes and making stuff prettier

Increased text sizes for readability, added rounded buttons and added background colors for all layouts.

ange-02 committed:
Fixes

penguoir commented on app/src/main/res/values/themes.xml:
```diff
@@ -1,16 +1,24 @@
 <resources xmlns:tools="http://schemas.android.com/tools">
     <!-- Base application theme. -->
-    <style name="Theme.ESPProject" parent="Theme.MaterialComponents.DayNight.NoActionBar">
+    <style name="Theme.MyApplication" parent="Theme.MaterialComponents.DayNight.NoActionBar">
```
This doesn't look right

penguoir replied:
In fact, it breaks the app for me

penguoir commented on app/src/main/res/layout/activity_profile.xml:
```diff
@@ -1,11 +1,244 @@
+        android:layout_height="wrap_content"
+        android:text="Achievements"
+        android:textColor="@color/black"
+        android:textSize="35sp"
+        android:textStyle="bold"
```
Can we extract this into components? e.g. a Title component rather than textSize and textStyle?

penguoir commented on app/src/main/res/layout/activity_profile.xml:
```diff
@@ -1,11 +1,244 @@
+                android:layout_height="wrap_content"
+                android:textSize="20sp"
+                android:text="You have saved"
+                android:textColor="@color/black"
```
Minor, but https://anthonyhobday.com/sideprojects/saferules/

penguoir commented on app/src/main/res/layout/activity_profile.xml:
```diff
@@ -1,11 +1,244 @@
+        android:textSize="20sp"
+        android:background="@drawable/rounded_button"
+        android:textColor="@color/white"
+        android:paddingHorizontal="50sp"
+        android:paddingVertical="15sp"
+        android:layout_marginTop="20sp"/>
```
Again, extract to reusable component.

penguoir commented on app/src/main/res/layout/dialog_add_ingredient.xml:
```diff
@@ -1,13 +1,90 @@
 
-    <Button
+    <TextView
+        android:id="@+id/add"
```
Aren't these ids global? I'd think we need more specific names

ange-02 replied:
Why has this become a textview?
Also, will change to dialogAddIngredientButton

penguoir commented on app/src/main/res/values/themes.xml:
```diff
@@ -1,16 +1,24 @@
+
+    <style name="Theme.MyApplication.AppBarOverlay" parent="ThemeOverlay.AppCompat.Dark.ActionBar" />
+
+    <style name="Theme.MyApplication.PopupOverlay" parent="ThemeOverlay.AppCompat.Light" />
```
Change MyApplication -> ESPProject

penguoir commented:
Just to be clear, I'm waiting for a commit on this branch that addresses the comments.

ange-02 committed:
Addressing comments

Haven't made the text styles into components yet, will do when I'm standardising everyones code.

penguoir commented:
Not all issues are fixed in the latest commit. The theme is still called MyApplication rather than ESPProject.

ange-02 committed:
Theme fix

ange-02 committed:
Theme fix #2

penguoir committed:
Squashed commit of Andrew's changes:

commit a8b3ee7dc22df704a7aca206892a3711f0c2d86c
Author: Andrew Grosuleac <amg208@bath.ac.uk>
Date:   Sun Feb 19 19:58:47 2023 +0000

    Theme fix #2

commit 5b38b3789ab8134fcd4a6e8838da5bdf335529b6
Author: Andrew Grosuleac <amg208@bath.ac.uk>
Date:   Sun Feb 19 19:55:22 2023 +0000

    Theme fix

commit 589ca8403baa164f0dd90b9f7531968e2647bdb7
Author: Andrew Grosuleac <amg208@bath.ac.uk>
Date:   Sun Feb 19 13:12:50 2023 +0000

    Addressing comments

    Haven't made the text styles into components yet, will do when I'm standardising everyones code.

commit 3ca666cca64906f21e31506b46acc9e856a362de
Author: Andrew Grosuleac <amg208@bath.ac.uk>
Date:   Wed Feb 15 13:48:21 2023 +0000

    Fixes

commit 9cf27ce90e345597bf557ee6b9510ad462b25ca9
Author: Andrew Grosuleac <amg208@bath.ac.uk>
Date:   Wed Feb 15 13:47:57 2023 +0000

    Fixes and making stuff prettier

    Increased text sizes for readability, added rounded buttons and added background colors for all layouts.

commit 4ce4ebc2fbeb71d2875b0f2261cf2f566052c9f9
Author: Andrew Grosuleac <amg208@bath.ac.uk>
Date:   Wed Feb 15 12:17:53 2023 +0000

    UI Pages and themes

    Contains initial UI design for main pantry screen, user settings, pantry users dialog and add ingredient dialog.
    Also, colors and themes.

penguoir committed:
Squashed commit of Ryan's changes:

commit 1e9774bc9f146ad538c61fe21f4c592c27c6bfeb
Author: Ragod12 <ryandsaug2002@gmail.com>
Date:   Thu Feb 16 12:17:56 2023 +0000

    ESPG16-133 Updated List datatype to ArrayList

    Changed all instances of List datatype to ArrayList

commit 373c8a500da06a15154ed7b0528beb37d3faee4d
Author: Ragod12 <ryandsaug2002@gmail.com>
Date:   Thu Feb 16 01:33:49 2023 +0000

    ESPG16-129 Created templates from UML diagram

    Used the UML diagram to create class template attributes ad methods. Needs more work I have highlighted TODOs

penguoir committed:
Squashed commit of more of Ryan's changes:

commit 11f55d5b8062670831d9a42304c09bc5045cac9c
Author: Ragod12 <ryandsaug2002@gmail.com>
Date:   Tue Feb 14 21:44:24 2023 +0000

    ESPG16-103 Updated Recipe Page (all recipes)

    Updated Recipe Page (all recipes) to be the page with a 'What's new scroller'.

commit bfd7f56f3785dfe767b900f7d66ee1a1a5063c03
Author: Ragod12 <ryandsaug2002@gmail.com>
Date:   Tue Feb 14 21:20:08 2023 +0000

    ESPG16-104 Update Cuisine List

    Updated Cuisine List (with search bar - see 3rd picture of recipe page prototypes)

commit 2b270c6e8a9562f0a03086d5f058e482a035519a
Author: Ragod12 <ryandsaug2002@gmail.com>
Date:   Tue Feb 14 20:36:33 2023 +0000

    ESPG16-103 Created Recipe page (all recipes)

    Created Recipe page (all recipes) and also created the recipebox layout for the list.

commit 75520ed19d0f06bb348b60bbb3f45ce1ff9c7ec5
Author: Ragod12 <ryandsaug2002@gmail.com>
Date:   Tue Feb 14 20:35:10 2023 +0000

    ESPG16-104 Create Recipe page (each cuisine)

    Created xml file for Recipe page (each cuisine)

penguoir committed:
Squashed commit of Faraz's changes:

commit 475522fbedbfaa215a92c9d29e02bd9a05a298bf
Author: farazovo <farazovo@gmail.com>
Date:   Tue Feb 14 20:55:05 2023 +0000

    minor fix to a line

commit 3dcff21215bb1b4d6d8b7e43ed64f40ef295adf2
Author: farazovo <farazovo@gmail.com>
Date:   Tue Feb 14 20:22:06 2023 +0000

    pages + gradel upgrade

    added a outline file called back to add outlines to text boxs
    put the code the fragment_recipe menu for menu
    put code for join/create panty in dialog create pantry
    (unsure if these locations are right)
    (if there are any more related fragments that i need to do lmk should be fast)

penguoir committed:
Squashed commit of Andy's changes:

commit 1a717467ce705ea87af5d7aebd1b4e35fd8441d3
Author: 4ndyK-C <andrewkerrchin@gmail.com>
Date:   Fri Feb 17 14:39:06 2023 +0000

    Files for Shopping list and Selected items

farazovo committed:
added Nav file

farazovo committed:
partial connect jav to xml

ionaclowes committed:
connect jav to xml

ionaclowes committed:
added to nav

farazovo committed:
Update misc.xml

farazovo committed:
Revert "Update misc.xml"

This reverts commit 9bc8c57baa7cfcedd72523ddb6f07fa40c534686.

farazovo committed:
names recipie menu

farazovo committed:
Update RecipeMenuFragment.java

farazovo committed:
WHY WONT MY EMULATOR WORKK

farazovo committed:
recipe menu to recipie nav WORKING

now that works just gotta copy to every page

ange-02 committed:
Added appname

ange-02 committed:
Fixed calendar

ionaclowes committed:
linked pages

farazovo committed:
Nav Done Dialog pages now

nearly done just have to figure out with dialogs

ekc30 committed:
ESPG16-91

ange-02 committed:
trying images

ange-02 committed:
Revert "trying images"

This reverts commit 60beb4075adb3510fba9c518ca6d793137b25d73.

farazovo committed:
All Nav expect Bar done

farazovo committed:
nav bar donee

dosent work for profile due to need to ask to change profile to a fragment, also needs aesthetic changes

ange-02 committed:
ESPG16-142 recipeFragment

Created a page for recipeFragment

ange-02 committed:
very important image uploaded

as title suggests
very important
(:

ekc30 committed:
Merge remote-tracking branch 'origin/Faraz-UI-pages-recipe-menu-and-pantry-join' into emi_branch

ange-02 committed:
Created lay file for ingredient and method recycler items

ekc30 committed:
Update misc.xml

ekc30 committed:
Merge remote-tracking branch 'origin/Proto+Nav' into emi_branch

ekc30 committed:
Merge Faraz and Emi branches

Merging the UI into one branch

ange-02 committed:
Recycler item for ingredient and method

ange-02 committed:
standardised colors

as title

ange-02 committed:
Merge remote-tracking branch 'origin/emi_branch' into andrews_ui
